catalog/transformer/catalog-transformer-streaming-lib/src/main/java/org/codice/ddf/transformer/xml/streaming/lib/SaxEventHandlerDelegate.java

```diff
@@ -0,0 +1,378 @@
+
+    private Map<String, List<SaxEventHandler>> eventHandlerLookup = new HashMap<>();
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SaxEventHandlerDelegate.class);
+
+    private MetacardType metacardType = BasicTypes.BASIC_METACARD;
+
+    public SaxEventHandlerDelegate() {
+        try {
+            parser = XMLReaderFactory.createXMLReader();
+        } catch (Exception e) {
+            LOGGER.debug(
+                    "Exception thrown during creation of SaxEventHandlerDelegate. Probably caused by one of the setFeature calls",
+                    e);
+        }
+    }
+
+    public SaxEventHandlerDelegate(SaxEventHandler... eventHandlers) {
+        this();
+        this.eventHandlers = Arrays.asList(eventHandlers);
+    }
+
+    public SaxEventHandlerDelegate(List<SaxEventHandler> eventHandlers) {
+        this();
+        this.eventHandlers = eventHandlers;
+    }
+
+    /**
+     * Takes in an {@link InputStream} returns a {@link Metacard},
+     * populated with all the {@link Attribute}s parsed by the {@link SaxEventHandlerDelegate#eventHandlers}
+     *
+     * @param inputStream an XML document that can be parsed into a Metacard
+     * @return a {@link Metacard},
+     * populated with all the {@link Attribute}s parsed by the {@link SaxEventHandlerDelegate#eventHandlers}
+     * @throws CatalogTransformerException
+     */
+    public Metacard read(InputStream inputStream) throws CatalogTransformerException {
+        configureEventHandlerLookup();
+
+        /*
+         * Create a new MetacardImpl with the proper MetacardType
+         */
+        Metacard metacard = new MetacardImpl(metacardType);
+
+        try {
+            InputSource newStream = new InputSource(new BufferedInputStream(inputStream));
+
+            /*
+             * Set the parser's ContentHandler to this delegate, which ensures the delegate receives all
+             * parse events that should be handled by a SaxEventHandler (startElement, endElement, characters, startPrefixMapping, etc)
+             * Set the parser's ErrorHandler to be a new InputTransformerHandler
+             */
+            parser.setContentHandler(this);
+            InputTransformerErrorHandler errorHandler = new InputTransformerErrorHandler();
+            parser.setErrorHandler(errorHandler);
+            parser.parse(newStream);
+
+            /*
+             * If any major errors occur during parsing, print them out
+             */
+            String errorsAndWarnings = errorHandler.getParseWarningsErrors();
+            if (!(errorsAndWarnings == null || errorsAndWarnings.isEmpty())) {
+                LOGGER.warn(errorHandler.getParseWarningsErrors());
+            }
+
+        } catch (IOException | SAXException e) {
+            LOGGER.debug("Exception thrown during parsing of inputStream", e);
+            throw new CatalogTransformerException("Could not properly parse metacard", e);
+        }
+
+        /*
```

Comment:
This feels like two methods to me. The first method parses the XML and the event handlers accumulate attributes.  The second methods pick up at `for (SaxEventHandler...` and populates the metacard with the accumulated attributes.